.github/workflows/release.yml

```diff
@@ -1,68 +1,105 @@
 name: Release
 
 on:
+  workflow_dispatch:
   push:
-    branches: [ master ]
+    branches: [master]
 
 jobs:
-  build:
-    name: Build and Release
+  test:
+    name: Test
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
 
       - uses: actions/setup-go@v4
         with:
-          go-version: '^1.20'
+          go-version: "^1.20"
+      - name: Go Linter
+        run: docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.52.0 golangci-lint run -v -E gofmt --timeout=5m
+
+      - name: Unit Tests
+        run: go test ./...
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    needs: test
+    if: github.event_name == 'workflow_dispatch'
+
+    outputs:
+      git_tag: ${{ steps.semantic_release_info.outputs.git_tag }}
+      version: ${{ steps.semantic_release_info.outputs.version }}
+      notes: ${{ steps.semantic_release_info.outputs.notes }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "^1.20"
 
       - name: Gets release info
         id: semantic_release_info
         uses: jossef/action-semantic-release-info@v2.1.0
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
doesn't make sense to split it if used only in `build`

Reply:
Splitting it allow me to decide to run the `build` job or not. If it will be part of the `build` job, it will require me to add an `if` to every step in `build`.